base/array.jl

```diff
@@ -142,9 +142,7 @@ julia> a = Base.vect(UInt8(1), 2.5, 1//2)
 """
 function vect(X...)
     T = promote_typeof(X...)
-    #T[ X[i] for i=1:length(X) ]
-    # TODO: this is currently much faster. should figure out why. not clear.
-    return copyto!(Vector{T}(undef, length(X)), X)
+    return getindex(T, X...)
```

Comment:
The `getindex` just reads a bit weird to me here, so I would write it as:
```suggestion
    return T[X...]
```